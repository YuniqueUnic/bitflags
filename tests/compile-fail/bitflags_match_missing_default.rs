bitflags::bitflags! {
    #[derive(Debug)]
    /// Example Flags
    pub struct Flags: u32 {
        /// A
        const A = 0b0000_0001;
        /// B
        const B = 0b0000_0010;
        /// C
        const C = 0b0000_0100;
        /// ABC
        const ABC = Flags::A.bits() | Flags::B.bits() | Flags::C.bits();

        const _ = !0;
    }
}

fn main() {
    bitflags::bitflags_match!(Flags::A | Flags::B, {
        Flags::A => (),
    });
}
