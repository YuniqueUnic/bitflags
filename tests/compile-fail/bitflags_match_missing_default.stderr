error: no rules expected `}`
  --> tests/compile-fail/bitflags_match_missing_default.rs:19:5
   |
19 | /     bitflags::bitflags_match!(Flags::A | Flags::B, {
20 | |         Flags::A => (),
21 | |     });
   | |______^ no rules expected this token in macro call
   |
note: while trying to match meta-variable `$t:tt`
  --> src/lib.rs
   |
   |     ($operation:expr, { $pattern:expr => $body:expr , $($t:tt)+ }) => {
   |                                                         ^^^^^
   = note: this error originates in the macro `bitflags::bitflags_match` (in Nightly builds, run with -Z macro-backtrace for more info)
